geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/spark/reproject/ReprojectWrapper.scala

```diff
@@ -48,49 +62,105 @@ object ReprojectWrapper {
     schema: String,
     returnedMetadata: String,
     destCRS: String,
+    layoutExtent: java.util.Map[String, Double],
+    tileLayout: java.util.Map[String, Int],
+    returnedResampleMethod: String,
     matchLayerExtent: Boolean
   ): (Int, (JavaRDD[Array[Byte]], String), String) =
     keyType match {
       case "SpatialKey" => {
         val metadataAST = returnedMetadata.parseJson
         val metadata = metadataAST.convertTo[TileLayerMetadata[SpatialKey]]
-        val layout = Right(metadata.layout)
+        val layout = Right(LayoutDefinition(layoutExtent.toExtent, tileLayout.toTileLayout))
 
-        reprojectRDD[SpatialKey](returnedRDD, schema, metadata, destCRS, layout, matchLayerExtent)
+        reprojectRDD[SpatialKey](returnedRDD, schema, metadata, destCRS, layout, matchLayerExtent, returnedResampleMethod)
       }
       case "SpaceTimeKey" => {
```

Comment:
This check is done all over the file and it looks like it can just happen in `private def reprojectRDD`